pkg/cluster/cluster.go

```diff
@@ -177,39 +177,50 @@ func New(log log.Logger, reg prometheus.Registerer, clusterEnabled bool, listenA
 		return nil, err
 	}
 
-	// Attempt to start the Node by connecting to the peers in gossipConfig.
-	// If we cannot connect to any peers, fall back to bootstrapping a new
-	// cluster by ourselves.
-	err = gossipNode.Start()
-	if err != nil {
-		level.Debug(log).Log("msg", "failed to connect to peers; bootstrapping a new cluster")
-		gossipConfig.JoinPeers = nil
-		err = gossipNode.Start()
+	return &Clusterer{Node: gossipNode}, nil
+}
+
+// Start starts the node.
+// For the localNode implementation, this is a no-op.
+// For the gossipNode implementation, Start will attempt to connect to the
+// configured list of peers; if this fails it will fall back to bootstrapping a
+// new cluster of its own.
+func (c *Clusterer) Start() error {
+	switch node := c.Node.(type) {
```

Comment:
Do we need the `Clusterer` struct? Could we instead add a `Start()` to the `Node`? You could have a no-op implementation on `localNode` and the full one on `GossipNode` and no need for a type assertion.

Reply:
That was my initial approach, but it didn't feel great to extend the Node interface with custom logic, since all of its other methods are direct wrappers over `ckit.Node`.

For the other use-cases, we could maybe remove the indirection layer and work directly with the embedded Node, but for now I'd like to still flesh it out.